src/ui/public/es.js

```diff
@@ -9,38 +9,39 @@ import 'elasticsearch-browser';
 import _ from 'lodash';
 import uiModules from 'ui/modules';
 
-let es; // share the client amongst all apps
+const plugins = [function (Client, config) {
+  // esFactory automatically injects the AngularConnector to the config
+  // https://github.com/elastic/elasticsearch-js/blob/master/src/lib/connectors/angular.js
+  class CustomAngularConnector extends config.connectionClass {
+    request = _.wrap(this.request, function (request, params, cb) {
+      if (String(params.method).toUpperCase() === 'GET') {
+        params.query = _.defaults({ _: Date.now() }, params.query);
+      }
+
+      return request.call(this, params, cb);
+    });
+  }
+
+  config.connectionClass = CustomAngularConnector;
+}];
+
 uiModules
   .get('kibana', ['elasticsearch', 'kibana/config'])
-  .service('es', function (esFactory, esUrl, $q, esApiVersion, esRequestTimeout) {
-    if (es) return es;
-
-    es = esFactory({
+  .service('es', function (esFactory, esUrl, esApiVersion, esRequestTimeout) {
```

Comment:
Can you add some comments explaining the difference between es and esAdmin.  If I'm understanding this correctly, es should be used to fetch data, while esAdmin should be used to fetch kibana stuff (like saved object meta data)?  So esAdmin will contain the .kibana index, and es will have the data stuff?  Maybe it should be called esData to better differentiate.  

It looks like everything got a bit more complicated now because we have to be aware of which is the right service to use.  I suppose there is no way to abstract that out so it would pick the right service?

When tribe node is off, do es and esAdmin point to the same cluster?  If so, it seems like it would be very easy for someone developing in non-tribe mode to break tribe mode without realizing it if they pick the wrong service, because either es or esAdmin would work in non-tribe node land.  Or I could be completely wrong, if that isn't how it works.